File: scala/etl/src/main/scala/Etl.scala
import scala.collection.mutable

object Etl {
  def transform(map: Map[Int, Seq[String]]): Map[String, Int] = {
    val cMap = FlatMap[String, Int]()
    for (value: (Int, Seq[String]) <- map) {
      for (subValue: String <- value._2) {
        cMap += (subValue.toLowerCase -> value._1)
      }
    }
    cMap.toMap
  }
}

File: typescript/etl/etl.ts
function transform(value: { [key: string]: string[] }): {
  [key: string]: number;
} {
  let result: {
    [key: string]: number;
  } = {};
  Object.keys(value).forEach((key) => {
    value[key].forEach((letter) => {
      result[letter.toLowerCase()] = Number(key);
    });
  });
  return result;
}

export default transform;
